server/application/terminal.go

```diff
@@ -227,6 +227,7 @@ func (s *terminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Failed to start terminal session", http.StatusBadRequest)
 		return
 	}
+	defer session.Close()
 	defer session.Done()
```

Comment:
I'm not sure if the order of `Close()` and `Done()` is important, but `defer` is executed in the opposite order than they are defined, i.e. with these two `defer`, `Done()` would be called before `Close()` - the previous behavior was to first call `Close()`, then `Done()`.